lib/registry.js

```diff
@@ -207,6 +219,16 @@ function formatLabels(labels) {
 	);
 }
 
+function flattenSharedLabels(labels) {
+	if (labels.__flattened_internal) {
+		return labels.__flattened_internal;
+	}
+
+	const formattedLabels = formatLabels(labels);
+	const flattened = formattedLabels.join(',');
+	Object.defineProperty(labels, '__flattened_internal', { value: flattened });
```

Comment:
I *think* you could use a WeakMap here instead of defining a property.

```diff
diff --git a/lib/registry.js b/lib/registry.js
index 2169b66..01b0f8e 100644
--- a/lib/registry.js
+++ b/lib/registry.js
@@ -219,14 +219,17 @@ function formatLabels(labels) {
 	);
 }
 
+const cache = new WeakMap();
+
 function flattenSharedLabels(labels) {
-	if (labels.__flattened_internal) {
-		return labels.__flattened_internal;
+	const cached = cache.get(labels);
+	if (cached) {
+		return cached;
 	}
 
 	const formattedLabels = formatLabels(labels);
 	const flattened = formattedLabels.join(',');
-	Object.defineProperty(labels, '__flattened_internal', { value: flattened });
+	cache.set(labels, flattened);
 	return flattened;
 }
 function escapeLabelValue(str) {
```

Reply:
Oh, that's a good idea. Let me fix up the changelog and make that change tomorrow morning.